### PR TITLE
Promote math constants `Kokkos::{Experimental -> numbers}` and add missing variables without the `_v` suffix

### PR DESCRIPTION
**Pull request by dalg24**
Rational: There is no good reason to keep them in `Kokkos::Experimental::` since it is just a backport of the C++20 facility.  I opted for the `Kokkos::numbers` namespace for consistency with the C++ standard.  Let me know if you want to capitalize the "N".

Drive-by changes:
* add missing math constant variables without the `_v` suffix

### CODE REVIEW

**Review by Rombur — APPROVED** [review]

File: core/src/impl/Kokkos_QuadPrecisionMath.hpp [hunk 1]
```diff
@@ -208,8 +208,7 @@ inline bool signbit(__float128 x) { return ::signbitq(x); }
 //</editor-fold>
 
 //<editor-fold desc="Mathematical constants __float128 specializations">
-namespace Kokkos {
-namespace Experimental {
+namespace Kokkos::numbers {
```
Comment (ldh4) [inline core/src/impl/Kokkos_QuadPrecisionMath.hpp:211]: I believe every Kokkos namespace starts with a capital letter (`Kokkos`, `Impl`, `Experimental`, `Test`, etc). I think it'll be better if we keep that consistent here as well.

**Comment by crtrott** [comment]
Lets discuss the namespace question today in the meeting.

File: core/src/impl/Kokkos_QuadPrecisionMath.hpp (hunk 1 above)
Comment (dalg24) [reply core/src/impl/Kokkos_QuadPrecisionMath.hpp:211]: This was discussed at the developer meeting.  Consensus was for `Kokkos::numbers` lowercase for consistency with C++20.

**Review by ldh4 — APPROVED** [review]

**Comment by dalg24** [comment]
HIP failure was my rebooting the new machine

**Pull request merged**